perf/change_point_detector.go

```diff
@@ -0,0 +1,74 @@
+package perf
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/evergreen-ci/cedar"
+	"io"
+	"net/http"
+)
+
+type Detector interface {
+	DetectChanges([]float64) ([]ChangePoint, error)
+}
+
+type ChangePoint struct {
+	Index     int
+	Algorithm Algorithm
+}
+
+type Algorithm struct {
+	Name          string
+	Version       string
+	Configuration map[string]interface{}
+}
+
+type signalProcessingClient struct {
+	execute    func(method, url string, body io.Reader) (*http.Response, error)
+}
+
+
+func NewDefaultDetector() Detector {
+	return NewDetector(http.DefaultClient.Do, http.NewRequest, cedar.GetEnvironment().GetConf().SignalProcessingURI, cedar.GetEnvironment().GetConf().SignalProcessingToken)
+}
+
+func NewDetector(requestExecutor func(req *http.Request) (*http.Response, error), requestMaker func(method, url string, body io.Reader) (*http.Request, error), baseURL string, token string) Detector {
+	executor := func(method, url string, body io.Reader) (*http.Response, error) {
+		req, err := requestMaker("POST", baseURL+"/change_points/detect", body)
```

Comment:
this function isn't defined?

Reply:
`requestMaker`? I think it's passed in to `NewDetector` as an arg. There is probably a better way to set this up in a way that can be mocked.

Reply:
I would just implement types and factor the methods on that so that you don't really need to mock things? Having indirection through an interface and a function type is overkill and just makes it really hard to figure out what's going on, and you can get almost all of the coverage you want by factoring you code into clearer units.